projects/igniteui-angular/src/lib/services/exporter-common/base-export-service.ts

```diff
@@ -208,72 +210,85 @@ export abstract class IgxBaseExporter {
             return record;
         });
 
-        this.exportGridRecordsData(records, options);
+        this.exportGridRecordsData(records);
     }
 
-    private exportGridRecordsData(records: IExportRecord[], options: IgxExporterOptionsBase) {
-        if (options === undefined || options === null) {
-            throw Error('No options provided!');
-        }
-
-        if (!this._columnList || this._columnList.length === 0) {
+    private exportGridRecordsData(records: IExportRecord[], grid?: IgxGridBaseDirective) {
+        if (this._ownersMap.size === 0) {
             const recordsData = records.map(r => r.data);
             const keys = ExportUtilities.getKeysFromData(recordsData);
-            this._columnList = keys.map((k) => ({ header: k, field: k, skip: false }));
-            this._columnWidthList = new Array<number>(keys.length).fill(DEFAULT_COLUMN_WIDTH);
+            const columns = keys.map((k) => ({ header: k, field: k, skip: false }));
+            const columnWidths = new Array<number>(keys.length).fill(DEFAULT_COLUMN_WIDTH);
+
+            const mapRecord: IColumnList = {
+                columns,
+                columnWidths,
+                indexOfLastPinnedColumn: -1
+            };
+
+            this._ownersMap.set(DEFAULT_OWNER, mapRecord);
         }
 
-        let skippedPinnedColumnsCount = 0;
-        let columnsWithoutHeaderCount = 1;
-        this._columnList.forEach((column, index) => {
-            if (!column.skip) {
-                const columnExportArgs = {
-                    header: !ExportUtilities.isNullOrWhitespaces(column.header) ?
-                        column.header :
-                        'Column' + columnsWithoutHeaderCount++,
-                    field: column.field,
-                    columnIndex: index,
-                    cancel: false,
-                    skipFormatter: false
-                };
-                this.columnExporting.emit(columnExportArgs);
+        for (const [key, mapRecord] of this._ownersMap) {
+            let skippedPinnedColumnsCount = 0;
+            let columnsWithoutHeaderCount = 1;
+            let indexOfLastPinnedColumn = mapRecord.indexOfLastPinnedColumn;
+
+            mapRecord.columns.forEach((column, index) => {
+                if (!column.skip) {
+                    const columnExportArgs: IColumnExportingEventArgs = {
+                        header: !ExportUtilities.isNullOrWhitespaces(column.header) ?
+                            column.header :
+                            'Column' + columnsWithoutHeaderCount++,
+                        field: column.field,
+                        columnIndex: index,
+                        cancel: false,
+                        skipFormatter: false,
+                        owner: key === DEFAULT_OWNER ? grid : key
```

Comment:
We forgot to replace the `owner` with `grid`.
```suggestion
                        grid: key === DEFAULT_OWNER ? grid : key
```